File: frontend/src/app/models/alert.ts
import { AlertRuleOperator } from './alert-rule';

export enum AlertStatus {
  Active = 'Active',
  Cleared = 'Cleared',
}

export interface Alert {
  id: number;
  alertRuleId: number;
  ruleName: string;
  deviceId: number;
  deviceName: string;
  hardwareId: string;
  telemetryKey: string;
  unit: string | null;
  value: number;
  thresholdValue: number;
  operator: AlertRuleOperator;
  status: AlertStatus;
  triggeredAt: string;
  clearedAt: string | null;
}

export interface AlertEvent {
  id: number;
  alertRuleId: number;
  ruleName: string;
  deviceId: number;
  hardwareId: string;
  telemetryKey: string;
  unit: string | null;
  value: number;
  thresholdValue: number;
  operator: AlertRuleOperator;
  status: AlertStatus;
  timestamp: string;
}

File: backend/src/MicroclimateIotSystem.Domain/Entities/Alert.cs
using MicroclimateIotSystem.Domain.Abstractions;
using MicroclimateIotSystem.Domain.Enums;

namespace MicroclimateIotSystem.Domain.Entities;

public class Alert : BaseEntity
{
    public int AlertRuleId { get; set; }
    public int DeviceId { get; set; }
    public string HardwareId { get; set; }
    public string TelemetryKey { get; set; }
    public string? Unit { get; set; }
    public double Value { get; set; }
    public double ThresholdValue { get; set; }
    public AlertRuleOperator Operator { get; set; }
    public AlertStatus Status { get; set; }
    public DateTime TriggeredAt { get; set; }
    public DateTime? ClearedAt { get; set; }

    public AlertRule AlertRule { get; set; }
    public Device Device { get; set; }
}